### PR TITLE
Move error logging to use diags.log instead of error.log

### PR DESCRIPTION
**Pull request by bryancall**
Consistently using Error() instead of sometimes using Log:error()

### CODE REVIEW

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -4175,28 +4175,28 @@ HttpSM::check_sni_host()
...
-                                 .extend(1)
-                                 .write('\0')
-                                 .data());
+              Error("%s", lbw()
+                            .clip(1)
+                            .print("No SNI for TLS request: connecting to {} for host='{}', returning a 403",
+                                   t_state.client_info.dst_addr, std::string_view{host_name, static_cast<size_t>(host_len)})
+                            .extend(1)
+                            .write('\0')
+                            .data());
```
Comment (bneradt) [inline proxy/http/HttpSM.cc:4184]: Maybe keep these connection-oriented messages in error.log? That's what I expect in error.log anyway: connection issues.

**Review by bneradt — CHANGES_REQUESTED** [review]
Thanks. I personally feel that error.log should just have connection (TCP or TLS) oriented issues. That's what I look to error.log for. Thus removing the "failed to log transaction..." message from error.log makes sense to me. But should the other connection related issues stay in error.log?

**Comment by bneradt** [comment]
[approve ci fedora]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by maskit** [comment]
I think we should mark this as incompatible and land this on 10-Dev branch first. Some people may watch error.log for error logs but not diags.log.

**Pull request closed without merging**